macros/src/trans.rs

```diff
@@ -18,8 +19,15 @@ pub fn app(app: &App, ownerships: &Ownerships) -> TokenStream {
     ::trans::resources(app, ownerships, &mut root);
 
     root.push(quote! {
+        use #krate::entry;
+
+        #[entry]
+        fn main() -> ! {
+            main_impl();
+        }
+
         #[allow(unsafe_code)]
-        fn main() {
+        fn main_impl() -> ! {
```

Comment:
Can't you just add `#[entry]` on main without creating `main_impl`?

Reply:
Well, let's see, is this all right? 493f71943a0d983d636f89135a88dd141b5e80d5